[ES|QL] Update grammars

kibanamachine opened the pull request:
This PR updates the ES|QL grammars (lexer and parser) to match the latest version in Elasticsearch.


## COMPLETION command update
```
completionCommand
    : COMPLETION (targetField=qualifiedName ASSIGN)? prompt=primaryExpression WITH inferenceId=identifierOrParameter
```

* No longer parse `COMPLETION ... AS <targetField>`
* Add parsing support for  `COMPLETION <targetField> = <prompt>`
* Allow `=` to be used within COMPLETION
* Update autocomplete and validation accordingly.

https://github.com/user-attachments/assets/6884bde4-fc14-4adb-9187-8e1ca125c52e

## RENAME
```
renameClause:
    oldName=qualifiedNamePattern AS newName=qualifiedNamePattern
    | newName=qualifiedNamePattern ASSIGN oldName=qualifiedNamePattern
```
Now renames support an ASSIGN expression.
* Support assign in the autocomplete.
* Support assign in the rename expression.
* Assign is treated in the same way as `AS`, only difference is that the params are reversed.

https://github.com/user-attachments/assets/3ccdaa62-bba5-4221-8f93-8b9b6e761a1e


## SAMPLE
```
sampleCommand
    : DEV_SAMPLE probability=decimalValue
```

* It does not parse seed value anymore.
* Cleaned tests. 

## FORK
* Fixed failing test
* Couldn't make CHANGE_POINT work within FORK, so didn't pushed anything related to supporting the commands within it.

<!--ONMERGE {"backportTargets":["8.19"]} ONMERGE-->

kibanamachine committed:
Update ES|QL grammars

sddonne committed:
fix: completion tests

sddonne committed:
fix: completion parser tests

sddonne committed:
fix: = validation within COMPLETION

sddonne committed:
fix: completion validation tests

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308736)
* Commit: 38ead95b2ebd73cabb0418b50e45cb1e35272054

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/308736#01977833-7af8-48e2-a23a-569e63f58b3f)

### History
* :broken_heart: [Build #308616](https://buildkite.com/elastic/kibana-pull-request/builds/308616) failed 81a3db447591c871c92ebe0f5d1cc7e322a21ea0

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308736","number":308736,"commit":"38ead95b2ebd73cabb0418b50e45cb1e35272054"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308616","number":308616,"commit":"81a3db447591c871c92ebe0f5d1cc7e322a21ea0"}],"number":308736}
buildkite-pr-comment-->

sddonne committed:
support assign in the rename command

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308828)
* Commit: fdf0034bc9c2b5525217e12267d6389ed2d90def

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/308828#019778e1-6d5b-4545-9b78-95db82effa7e)

### History
* :broken_heart: [Build #308736](https://buildkite.com/elastic/kibana-pull-request/builds/308736) failed 38ead95b2ebd73cabb0418b50e45cb1e35272054
* :broken_heart: [Build #308616](https://buildkite.com/elastic/kibana-pull-request/builds/308616) failed 81a3db447591c871c92ebe0f5d1cc7e322a21ea0

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308828","number":308828,"commit":"fdf0034bc9c2b5525217e12267d6389ed2d90def"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308736","number":308736,"commit":"38ead95b2ebd73cabb0418b50e45cb1e35272054"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308616","number":308616,"commit":"81a3db447591c871c92ebe0f5d1cc7e322a21ea0"}],"number":308828}
buildkite-pr-comment-->

sddonne committed:
fix: take out seed from SAMPLE

sddonne committed:
clean

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308860)
* Commit: 7021fb709412ee4e78f63e675ee25b15b3af8782

### Failed CI Steps
* [Jest Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/308860#01977928-dbb5-42e9-b913-93c292452a49)
* [Jest Tests #7](https://buildkite.com/elastic/kibana-pull-request/builds/308860#01977928-dbb7-46aa-88a7-1302ecf897e9)
* [Jest Tests #7](https://buildkite.com/elastic/kibana-pull-request/builds/308860#01977955-0dd5-43cc-a4a9-5a72e22657fc)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308860#01977928-dbb5-42e9-b913-93c292452a49) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308860/jobs/01977928-dbb5-42e9-b913-93c292452a49/artifacts/01977954-72e7-4417-98b3-22821fdb9b70) Jest Tests #<span></span>1 / autocomplete Invoke trigger kind (all commands) FROM index1 | RENAME f/
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308860#01977928-dbb5-42e9-b913-93c292452a49) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308860/jobs/01977928-dbb5-42e9-b913-93c292452a49/artifacts/01977954-72e8-4596-8882-da04f590cc1a) Jest Tests #<span></span>1 / autocomplete Invoke trigger kind (all commands) FROM index1 | RENAME field A/
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308860#01977928-dbb5-42e9-b913-93c292452a49) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308860/jobs/01977928-dbb5-42e9-b913-93c292452a49/artifacts/01977954-72e7-4ed2-98f0-09e51ca67c98) Jest Tests #<span></span>1 / autocomplete.suggest RENAME suggests fields
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308860#01977928-dbb7-46aa-88a7-1302ecf897e9) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308860/jobs/01977928-dbb7-46aa-88a7-1302ecf897e9/artifacts/01977954-e322-4177-8ae0-ee9fd5a5fe34) Jest Tests #<span></span>7 / can create a SAMPLE command
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308860#01977955-0dd5-43cc-a4a9-5a72e22657fc) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308860/jobs/01977955-0dd5-43cc-a4a9-5a72e22657fc/artifacts/0197795b-3a1a-4aa5-9c29-1f5f0dbf436a) Jest Tests #<span></span>7 / can create a SAMPLE command
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308860#01977928-dbb5-42e9-b913-93c292452a49) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308860/jobs/01977928-dbb5-42e9-b913-93c292452a49/artifacts/01977954-72e7-4737-a5a3-4e9fe56de5c7) Jest Tests #<span></span>1 / COMPLETION adds the given targetField as field, when AS option is specified
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308860#01977928-dbb7-46aa-88a7-1302ecf897e9) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308860/jobs/01977928-dbb7-46aa-88a7-1302ecf897e9/artifacts/01977954-e324-4b3c-9c55-14e7f03fae2c) Jest Tests #<span></span>7 / single line query commands COMPLETION from multiline
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308860#01977955-0dd5-43cc-a4a9-5a72e22657fc) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308860/jobs/01977955-0dd5-43cc-a4a9-5a72e22657fc/artifacts/0197795b-3a1a-4848-a1ac-50695c411bb3) Jest Tests #<span></span>7 / single line query commands COMPLETION from multiline
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308860#01977928-dbb7-46aa-88a7-1302ecf897e9) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308860/jobs/01977928-dbb7-46aa-88a7-1302ecf897e9/artifacts/01977954-e323-4b64-8e6a-f0e86bb0a2ce) Jest Tests #<span></span>7 / single line query commands COMPLETION from single line
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308860#01977955-0dd5-43cc-a4a9-5a72e22657fc) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308860/jobs/01977955-0dd5-43cc-a4a9-5a72e22657fc/artifacts/0197795b-3a19-4922-89a8-5d8a924b6fa0) Jest Tests #<span></span>7 / single line query commands COMPLETION from single line
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308860#01977928-dbb7-46aa-88a7-1302ecf897e9) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308860/jobs/01977928-dbb7-46aa-88a7-1302ecf897e9/artifacts/01977954-e324-4212-a305-f5b407556246) Jest Tests #<span></span>7 / single line query commands SAMPLE from multiline
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308860#01977955-0dd5-43cc-a4a9-5a72e22657fc) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308860/jobs/01977955-0dd5-43cc-a4a9-5a72e22657fc/artifacts/0197795b-3a1a-410b-a724-33b98926b6bc) Jest Tests #<span></span>7 / single line query commands SAMPLE from multiline
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308860#01977928-dbb7-46aa-88a7-1302ecf897e9) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308860/jobs/01977928-dbb7-46aa-88a7-1302ecf897e9/artifacts/01977954-e325-4eb4-b81b-ab0f54afcc8a) Jest Tests #<span></span>7 / single line query commands SAMPLE from single line
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308860#01977955-0dd5-43cc-a4a9-5a72e22657fc) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308860/jobs/01977955-0dd5-43cc-a4a9-5a72e22657fc/artifacts/0197795b-3a1a-49c4-ad3f-bf332deeee24) Jest Tests #<span></span>7 / single line query commands SAMPLE from single line
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308860#01977928-dbb5-42e9-b913-93c292452a49) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308860/jobs/01977928-dbb5-42e9-b913-93c292452a49/artifacts/01977954-72e8-4470-8460-e8b8a86f0196) Jest Tests #<span></span>1 / validation command FORK ... (SUBCOMMAND ...) ... forwards syntax errors
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308860#01977928-dbb5-42e9-b913-93c292452a49) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308860/jobs/01977928-dbb5-42e9-b913-93c292452a49/artifacts/01977954-72e8-44ed-bd2b-1fbc832b2fd4) Jest Tests #<span></span>1 / validation logic Full validation performed rename from a_index | rename a => 2 errors, 0 warnings
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308860#01977928-dbb5-42e9-b913-93c292452a49) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308860/jobs/01977928-dbb5-42e9-b913-93c292452a49/artifacts/01977954-72e8-476f-bc43-1facfaebb8d5) Jest Tests #<span></span>1 / validation logic Full validation performed rename from a_index | rename textField => 1 errors, 0 warnings

### History
* :broken_heart: [Build #308828](https://buildkite.com/elastic/kibana-pull-request/builds/308828) failed fdf0034bc9c2b5525217e12267d6389ed2d90def
* :broken_heart: [Build #308736](https://buildkite.com/elastic/kibana-pull-request/builds/308736) failed 38ead95b2ebd73cabb0418b50e45cb1e35272054
* :broken_heart: [Build #308616](https://buildkite.com/elastic/kibana-pull-request/builds/308616) failed 81a3db447591c871c92ebe0f5d1cc7e322a21ea0

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308860","number":308860,"commit":"7021fb709412ee4e78f63e675ee25b15b3af8782"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308828","number":308828,"commit":"fdf0034bc9c2b5525217e12267d6389ed2d90def"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308736","number":308736,"commit":"38ead95b2ebd73cabb0418b50e45cb1e35272054"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308616","number":308616,"commit":"81a3db447591c871c92ebe0f5d1cc7e322a21ea0"}],"number":308860}
buildkite-pr-comment-->

sddonne committed:
fix: tests

sddonne committed:
fix: FORK test

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308965)
* Commit: 73462ca03edc81ddcd5a6aff4c6ad480e667face

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/951bc75b27116e5be51f1b0a8d262e7e89d58009) | [after](https://github.com/elastic/kibana/commit/73462ca03edc81ddcd5a6aff4c6ad480e667face) | diff |
| --- | --- | --- | --- |
| `kbnUiSharedDeps-srcJs` | 3.7MB | 3.7MB | +1.3KB |



### History
* :broken_heart: [Build #308860](https://buildkite.com/elastic/kibana-pull-request/builds/308860) failed 7021fb709412ee4e78f63e675ee25b15b3af8782
* :broken_heart: [Build #308828](https://buildkite.com/elastic/kibana-pull-request/builds/308828) failed fdf0034bc9c2b5525217e12267d6389ed2d90def
* :broken_heart: [Build #308736](https://buildkite.com/elastic/kibana-pull-request/builds/308736) failed 38ead95b2ebd73cabb0418b50e45cb1e35272054
* :broken_heart: [Build #308616](https://buildkite.com/elastic/kibana-pull-request/builds/308616) failed 81a3db447591c871c92ebe0f5d1cc7e322a21ea0

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308965","number":308965,"commit":"73462ca03edc81ddcd5a6aff4c6ad480e667face"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308860","number":308860,"commit":"7021fb709412ee4e78f63e675ee25b15b3af8782"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308828","number":308828,"commit":"fdf0034bc9c2b5525217e12267d6389ed2d90def"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308736","number":308736,"commit":"38ead95b2ebd73cabb0418b50e45cb1e35272054"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308616","number":308616,"commit":"81a3db447591c871c92ebe0f5d1cc7e322a21ea0"}],"number":308965}
buildkite-pr-comment-->

stratoula committed:
Merge branch 'main' into esql_grammar_sync_1750046849

stratoula reviewed (commented):
@sddonne thanx for trying to fix this. I found something a bug on completion and rename autocomplete. Can you take care of them please? Thanx!

vadimkibana commented on src/platform/packages/shared/kbn-esql-ast/src/visitor/contexts.ts:
```diff
@@ -61,7 +61,7 @@ const isRenameExpression = (
     parent.name === 'rename' &&
     node.type === 'option' &&
-    node.name === 'as'
+    (node.name === 'as' || node.name === '=')
```
And eventually rewrite `RENAME` to make `AS` in `RENAME` a binary expression, not an `option`.

sddonne replied:
This is not for `COMPLETION` in there `=` is parsed into a binary expression.

I added this change because `RENAME` clause has been updated too, now it can hold both `AS` and `=`, my intention was directed to update the renameClause and handle `AS` and `=` in a similar way, otherwise we would have a very different handling for similar things within the `RENAME` command itself.

I guess that for handling this optimally we should deeply modify `RENAME` and handle `AS` and `=` as binary-expression.

@vadimkibana @stratoula, wdyt about merging this change as is, handling `AS` and `=` similarly in the rename clause, and I can try opening a new PR with the refactor moving both to binary expression?

I'm afraid the change might be complex to put in this PR that also holds so many other things.

stratoula replied:
Sure if Vadim is ok, I am ok to do it on a follow up 👍

sddonne replied:
@vadimkibana is this similar to what you had in mind? 
https://github.com/elastic/kibana/pull/224292
(I need to manual test a bit more the pretty printer still)

sddonne replied:
so, effort is already made, so if @vadimkibana is ok with this solution https://github.com/elastic/kibana/pull/224292, I can push those changes into this PR

vadimkibana commented on src/platform/packages/shared/kbn-esql-ast/src/parser/factories/completion.ts:
```diff
@@ -10,18 +10,51 @@ import { ESQLAstCompletionCommand, ESQLSingleAstItem } from '../../types';
+    command.targetField = targetField;
+    command.args.push(assignment);
+  } else {
+    // When the user is typing a column as prompt i.e: | COMPLETION message^,
```
https://github.com/elastic/kibana/pull/224010#discussion_r2151587187

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/309164)
* Commit: c9bc2d4bd3b2e8fb296face2b07d386ecdb48c3e

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/d9a3e1f40093f85dda51676a1c5657fdd0c8a0b0) | [after](https://github.com/elastic/kibana/commit/c9bc2d4bd3b2e8fb296face2b07d386ecdb48c3e) | diff |
| --- | --- | --- | --- |
| `kbnUiSharedDeps-srcJs` | 3.7MB | 3.7MB | +1.3KB |



### History
* :green_heart: [Build #308965](https://buildkite.com/elastic/kibana-pull-request/builds/308965) succeeded 73462ca03edc81ddcd5a6aff4c6ad480e667face
* :broken_heart: [Build #308860](https://buildkite.com/elastic/kibana-pull-request/builds/308860) failed 7021fb709412ee4e78f63e675ee25b15b3af8782
* :broken_heart: [Build #308828](https://buildkite.com/elastic/kibana-pull-request/builds/308828) failed fdf0034bc9c2b5525217e12267d6389ed2d90def
* :broken_heart: [Build #308736](https://buildkite.com/elastic/kibana-pull-request/builds/308736) failed 38ead95b2ebd73cabb0418b50e45cb1e35272054
* :broken_heart: [Build #308616](https://buildkite.com/elastic/kibana-pull-request/builds/308616) failed 81a3db447591c871c92ebe0f5d1cc7e322a21ea0

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309164","number":309164,"commit":"c9bc2d4bd3b2e8fb296face2b07d386ecdb48c3e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308965","number":308965,"commit":"73462ca03edc81ddcd5a6aff4c6ad480e667face"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308860","number":308860,"commit":"7021fb709412ee4e78f63e675ee25b15b3af8782"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308828","number":308828,"commit":"fdf0034bc9c2b5525217e12267d6389ed2d90def"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308736","number":308736,"commit":"38ead95b2ebd73cabb0418b50e45cb1e35272054"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308616","number":308616,"commit":"81a3db447591c871c92ebe0f5d1cc7e322a21ea0"}],"number":309164}
buildkite-pr-comment-->

sddonne committed:
only suggest = after a column that does not exists

sddonne committed:
refactor: completion parser

sddonne commented on src/platform/packages/shared/kbn-esql-ast/src/parser/factories/completion.ts:
```diff
@@ -10,18 +10,51 @@ import { ESQLAstCompletionCommand, ESQLSingleAstItem } from '../../types';
+
+    command.prompt = prompt;
+    command.args.push(prompt);
+  }
```
Looks more clear 👍 , taken daa662d8c1169e1ce43aa34e4ed7e20f42406746

sddonne commented on src/platform/packages/shared/kbn-esql-ast/src/parser/__tests__/completion.test.ts:
```diff
@@ -198,6 +241,50 @@ describe('COMPLETION command', () => {
+
+      expect(completionCommand.prompt).toMatchObject({
+        type: 'unknown',
+        name: 'unknown',
```
Did not get this one, this node is meanly meant to understand that the parser found something but it cannot decide yet what it is, `unknown` seems more relatable to me, but for sure I might be missing something 🤔

vadimkibana replied:
It is good as it is now. I was just thinking in the future we could store the actual prompt value in the `.name` field. Currently, both, `.type` and `.name` are set to `"unknown"`, which is not very useful. Eventually we want to deprecate the `.text` field, so I was thinking if we stored the actual propmt text in `.name` it could be useful. But, I realize that is not something we do with other `{type: 'unknown'}` currently, so not something required for this PR.

stratoula replied:
okie

sddonne committed:
support new commands in fork

## Summary

Summarize your PR. If it involves visual changes include a screenshot or
gif.


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

---------

Co-authored-by: Stratoula <efstratia.kalafateli@elastic.co>

sddonne committed:
Merge remote-tracking branch 'upstream/main' into esql_grammar_sync_1750046849

sddonne committed:
Merge branch 'esql_grammar_sync_1750046849' of https://github.com/elastic/kibana into esql_grammar_sync_1750046849

sddonne committed:
Merge remote-tracking branch 'upstream/main' into esql_grammar_sync_1750046849

sddonne committed:
Fix AS suggestion after RENAME field a/

sddonne committed:
support = in rename query util

sddonne committed:
simplify

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/309782)
* Commit: 61e883c32dad15377e80c15c45c82ad8d2e6ac93

### Failed CI Steps
* [Jest Tests #17](https://buildkite.com/elastic/kibana-pull-request/builds/309782#01978262-0253-4b84-9cf3-92e6f02fcc15)
* [Jest Tests #17](https://buildkite.com/elastic/kibana-pull-request/builds/309782#01978286-b9ba-4ebc-8996-fd3517767b46)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/309782#01978262-0253-4b84-9cf3-92e6f02fcc15) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/309782/jobs/01978262-0253-4b84-9cf3-92e6f02fcc15/artifacts/01978286-aab5-449e-9f3d-a4663ba887d7) Jest Tests #<span></span>17 / autocomplete.suggest RENAME suggests = after a field that does not exist
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/309782#01978286-b9ba-4ebc-8996-fd3517767b46) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/309782/jobs/01978286-b9ba-4ebc-8996-fd3517767b46/artifacts/0197828b-4957-4e7c-a83c-bae90efa512c) Jest Tests #<span></span>17 / autocomplete.suggest RENAME suggests = after a field that does not exist

### History
* :broken_heart: [Build #309383](https://buildkite.com/elastic/kibana-pull-request/builds/309383) failed 2eb7072a0efa3cf26850783db6a8cb9448c51eab
* :broken_heart: [Build #309239](https://buildkite.com/elastic/kibana-pull-request/builds/309239) failed daa662d8c1169e1ce43aa34e4ed7e20f42406746
* :green_heart: [Build #309164](https://buildkite.com/elastic/kibana-pull-request/builds/309164) succeeded c9bc2d4bd3b2e8fb296face2b07d386ecdb48c3e
* :green_heart: [Build #308965](https://buildkite.com/elastic/kibana-pull-request/builds/308965) succeeded 73462ca03edc81ddcd5a6aff4c6ad480e667face
* :broken_heart: [Build #308860](https://buildkite.com/elastic/kibana-pull-request/builds/308860) failed 7021fb709412ee4e78f63e675ee25b15b3af8782

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309782","number":309782,"commit":"61e883c32dad15377e80c15c45c82ad8d2e6ac93"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309383","number":309383,"commit":"2eb7072a0efa3cf26850783db6a8cb9448c51eab"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309239","number":309239,"commit":"daa662d8c1169e1ce43aa34e4ed7e20f42406746"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309164","number":309164,"commit":"c9bc2d4bd3b2e8fb296face2b07d386ecdb48c3e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308965","number":308965,"commit":"73462ca03edc81ddcd5a6aff4c6ad480e667face"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308860","number":308860,"commit":"7021fb709412ee4e78f63e675ee25b15b3af8782"}],"number":309782}
buildkite-pr-comment-->

vadimkibana reviewed (approved):
LGTM, code review only. Lets address `AS` option in a separate PR.

sddonne committed:
simpler implementatiion

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/310002)
* Commit: f2836789f83dd8bb8b604bdf7d5022f6065dd973

### Failed CI Steps
* [FTR Configs #59](https://buildkite.com/elastic/kibana-pull-request/builds/310002#0197837a-ef58-4c38-8f0a-8e371dfe3bcf)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/310002#0197837a-ef58-4c38-8f0a-8e371dfe3bcf) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/310002/jobs/0197837a-ef58-4c38-8f0a-8e371dfe3bcf/artifacts/019783a4-3fef-4f1d-8d26-11790bb57340) FTR Configs #<span></span>59 / Rule execution logic API - Basic License/Essentials Tier @<span></span>ess @<span></span>serverless @<span></span>serverlessQA timestamp tests alerts generated from events with timestamp override field EQL should generate 2 alerts with @<span></span>timestamp

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/0126bf6af2728db42762a5be13d53609e33a1e9c) | [after](https://github.com/elastic/kibana/commit/f2836789f83dd8bb8b604bdf7d5022f6065dd973) | diff |
| --- | --- | --- | --- |
| `kbnUiSharedDeps-srcJs` | 3.7MB | 3.7MB | +1.7KB |
| `stackAlerts` | 27.5KB | 27.5KB | +67.0B |
| total |  |  | +1.8KB |



### History
* :broken_heart: [Build #309782](https://buildkite.com/elastic/kibana-pull-request/builds/309782) failed 61e883c32dad15377e80c15c45c82ad8d2e6ac93
* :broken_heart: [Build #309383](https://buildkite.com/elastic/kibana-pull-request/builds/309383) failed 2eb7072a0efa3cf26850783db6a8cb9448c51eab
* :broken_heart: [Build #309239](https://buildkite.com/elastic/kibana-pull-request/builds/309239) failed daa662d8c1169e1ce43aa34e4ed7e20f42406746
* :green_heart: [Build #309164](https://buildkite.com/elastic/kibana-pull-request/builds/309164) succeeded c9bc2d4bd3b2e8fb296face2b07d386ecdb48c3e
* :green_heart: [Build #308965](https://buildkite.com/elastic/kibana-pull-request/builds/308965) succeeded 73462ca03edc81ddcd5a6aff4c6ad480e667face

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310002","number":310002,"commit":"f2836789f83dd8bb8b604bdf7d5022f6065dd973"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309782","number":309782,"commit":"61e883c32dad15377e80c15c45c82ad8d2e6ac93"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309383","number":309383,"commit":"2eb7072a0efa3cf26850783db6a8cb9448c51eab"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309239","number":309239,"commit":"daa662d8c1169e1ce43aa34e4ed7e20f42406746"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309164","number":309164,"commit":"c9bc2d4bd3b2e8fb296face2b07d386ecdb48c3e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308965","number":308965,"commit":"73462ca03edc81ddcd5a6aff4c6ad480e667face"}],"number":310002}
buildkite-pr-comment-->

stratoula reviewed (approved):
This seems to work great now 🤞 Thanx for the patience Sebastian, it was a group of significant changes from ES

kibanamachine commented:
Starting backport for target branches: 8.19

https://github.com/elastic/kibana/actions/runs/15757823918

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.19|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/224561">](https://github.com/elastic/kibana/pull/224561)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->
